src/Components/userpages/chat.js

```diff
@@ -0,0 +1,52 @@
+import React from 'react'
+import Userbase from '../Layout/userbase'
+import io from 'socket.io-client';
+var socket = null
+class Chat extends React.Component{
+
+    constructor(props){
+        super(props)
+        this.state = {chatMessages:[]}
+
+    }
+    emitmsg=(e)=>{
+        e.preventDefault();
+        
+        
+        console.log('message',document.getElementById("m").value)
+        socket.emit('chat message',document.getElementById("m").value);
+        document.getElementById("m").value = null;
+       
+
+    }
+    componentDidMount (){
+        socket = io("http://localhost:3001")
+         console.log(socket)
+         socket.on('connect', function () {
+                console.log("connected on client")
+                
+                       })
+        socket.on('chat message', (msg)=>{
+            this.state.chatMessages.push(<div className='well'>{msg}</div>)
```

Comment:
stop pushing divs in state, only push the data and then iterate over it in render and display the jsx